segmentation/include/pcl/segmentation/extract_polygonal_prism_data.h

```diff
@@ -189,6 +200,9 @@ namespace pcl
       /** \brief A pointer to the input planar hull dataset. */
       PointCloudConstPtr planar_hull_;
 
+      /** \brief polygons indices vectors, as recieved from ConcaveHull */
+      std::vector<pcl::Vertices> rings_;
```

Comment:
Should this be a shared_ptr<vector>? Smart user could save a bunch of copies in L139, at cost of 2 dynamic allocations

Reply:
**This is a good question.** since ConcaveHull gives out `std::vector<pcl::Vertices>` and not a `std::vector<pcl::Vertices::Ptr>`, I cant see how to save the copies, because `make_shared` creates a copy anyways.
One alternative was to send the vector as refernce when using the `segment` method. But I prefered to do it this way.
In all of `pcl` `Vertices::Ptr` isn't used once.

Reply:
I would leave it this way (with const reference and copies), since it is done the same way in crop_hull.h